modules/lang-painless/src/main/java/org/elasticsearch/painless/FunctionRef.java

```diff
@@ -80,9 +85,45 @@
      * @param call the right hand side of a method reference expression
      * @param numCaptures number of captured arguments
      */
-    public FunctionRef(PainlessLookup painlessLookup, Class<?> expected, String type, String call, int numCaptures) {
-        this(expected, painlessLookup.getPainlessStructFromJavaClass(expected).functionalMethod,
-                lookup(painlessLookup, expected, type, call, numCaptures > 0), numCaptures);
+    public static FunctionRef getFunctionRef(PainlessLookup painlessLookup, Class<?> expected, String type, String call, int numCaptures) {
```

Comment:
I would name this maybe "resolveFromLookup" or something like that, since the purpose compared to the actor is to resolve a signature into a FunctionRef object by looking up in the lookup?

Reply:
Done.